Item

WhiteRabbit27 opened the pull request:
index, add, view single, view all, update and delete work with the following:
route('/', methods=['GET', 'POST'])
route('/<string:itemName>', methods=['GET', 'PUT', 'DELETE'])

If there is specific functionality you might need/want, let me know

Commit:
Item and controller started.

Commit:
quick save

Commit:
erge branch 'master' into item

codyleyhan committed:
created script for init db

Commit:
save before pull from master

codyleyhan committed:
fixed bugs in restaurant

Commit:
push for cody

Commit:
push for cody

Commit:
save before pull fromm master

Commit:
to silence the merge

codyleyhan committed:
fixed initdb.sh error

Commit:
ready for merge

Commit:
save

Commit:
get and dict not working

Commit:
/<string> ... 'GET' works

Commit:
/<string:itemName> request.method=='GET' works

Commit:
DELETE works

Commit:
add and delete item works

Commit:
add/del/view/viewall work

Commit:
updated everything except update

Commit:
all module working: i.e. index, add, view single, view all, update and delete all work

Commit:
all changes accepted and ready for pull request from master

Commit:
minor details added to 'clean-up :item:'

Commit:
formatting to index

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
 from flask import Blueprint, request, g, redirect, url_for, jsonify
 from datetime import datetime, timedelta
+import pprint
```
I dont think we need this line

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+    #index page
+    if request.method == 'GET':
+        return jsonify({
+            'legal routes': {
```
Don't want to give the user too much information,
also this route should return all items

WhiteRabbit27 replied:
wrt index: if user: dispay items, costs, descriptions.

codyleyhan replied:
Yeah, after the route is entirely nailed down we will update to auth

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+            check_item = get(name)
+            if check_item:
+                return jsonify(
+                    '%r already exists in your system.' % name,{
```
I think that we are going to go with this format:
```javascript
{
'status': 'error or success',
'message' : 'message',
'errors': {
    'key' : ['errors']
}
```

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+            item = Item(request.json['name'], request.json['cost'], request.json['description'])
+            db.session.add(item)
+            db.session.commit()
+            return jsonify(
```
Same as above here

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+            }), 201
+        return jsonify({'message': 'There was an error adding data', 'error': form.errors}), 400
+
+@itemMod.route('/<string:itemName>', methods=['GET', 'PUT', 'DELETE'])
```
we should use item id to avoid any kinds of sanitation problem so maybe

```python
@itemMod.route('/<int:itemId>', methods=['GET','PUT','DELETE'])
```

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+@itemMod.route('/<string:itemName>', methods=['GET', 'PUT', 'DELETE'])
+def update(itemName):
+    # Query all items in the database and return.
+    if itemName == 'all':
```
This should be in the index route

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+    item = get(itemName)
+    print(item)
+    if not item:
+        return jsonify('%r not found in database.' % itemName), 400
```
json message same as above

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+    # view a single Item
+    if request.method == 'GET':
+        item = a_dict(item)
+        return jsonify({
```
Same as above

codyleyhan commented on app/item/controller.py:
```diff
@@ -1,4 +1,104 @@
+
+    # update a single Item
+    if request.method == 'PUT':
+        try:
```
validate input before updating

codyleyhan commented on app/item/models/Item.py:
```diff
@@ -0,0 +1,26 @@
+        return item
+
+    def __repr__(self):
+        return '<Item %r $%f %r>' % (self.name, self.cost, self.description)
```
I think we should probably just do a name here and possibly id

codyleyhan commented on app/item/validators/ItemValidator.py:
```diff
@@ -0,0 +1,25 @@
+from wtforms import Form, StringField, IntegerField, FloatField, validators
+
+class ItemValidator(Form):
+    restaurant_number = IntegerField('Restaurant id', [
```
maybe update this to say id rather than number can cause confusion

codyleyhan commented on app/item/validators/ItemValidator.py:
```diff
@@ -0,0 +1,25 @@
+    ])
+
+    """
+    #description is an optional field
```
I think description should be a required field

codyleyhan reviewed (changes requested):
I put in some comments in the code review, you should be doing these for me too.  WE should have two eyes on all code that we write.  Also can we remove all the extra unneeded .pyc files.  you will have to remove them from the git cache also add *.pyc  and *.pyo to the .gitignore file

codyleyhan committed:
added restaurant delete

codyleyhan committed:
CRUD for restaurant done

codyleyhan committed:
added docs for the restaurant module

codyleyhan committed:
fixed spacing issues

Commit:
index updated to display all items

Commit:
route('/') up and running

Commit:
all returns updated

Commit:
all modules working. update updated.

Commit:
Merge branch 'restaurant' of https://github.com/codyleyhan/dinero

Commit:
Merge branch 'restaurant' of https://github.com/codyleyhan/dinero into restaurant

Commit:
get(arg) updated to handle str and int arguements

Commit:
update updates. does not check for duplicate items.

Commit:
TODO added in Item.py line 25. rmoved alternate and original __repr__ <>'s

Commit:
controller.py full functionality. tweaks to come

Commit:
update from :branch: item

Commit:
__repr__ updated and __str__ added with same value set as __repr__ except with '' instead of <>

Commit:
in ItemValidators: restaurant number changed to restaurant id

Commit:
Merge branch 'item' into itemMaster

__repr__ updated from "<Item %r %d %r>" to "<Item %d %r>" % (item.id, item.name)
__str__	 added. "Item %r %d" % (item.id, item.name). used for jsonify returns

update() Now checks whether there is an item in the database with the name of the item
	 to be updated. If there is an item with that name, then item exists exception
	 is raised. FUTURE: check for unique item/id match before updating

Commit:
Update filters items with same descriptions and different itemIds

Commit:
update updated. Next commit will have fully functioning update

Commit:
update returns all items that have the same name

Commit:
Update similar items found return algorithm

Commit:
Updates to Item():
    __repr__ format:
        changed from: <Item id, name>
        changed to:   <Item id, name, cost, description>
        reasoning: python documentation:
                   "Called by the repr() built-in function and by string conversions
                    (reverse quotes) to compute the "official" string representation
                    of an object. If at all possible, this should look like a valid
                    Python expression that could be used to recreate an object with
                    the same value (given an appropriate environment)."

    __str__ format:
        changed from: "Item id, name"
        changed to:   <Item id, name>
        reasoning: python documentation:
                   "Called by the str() built-in function and by the print statement
                    to compute the "informal" string representation of an object.
                    Updates to Update():

Updates to ::contorller.py:: Update():
    Update no updates an item if, and only if, there are no other items in the
    database containing the same name OR description. If Item objects containing
    either the same name or description (and containing a UNIQUE itemId), then
    items are captured and returned as a tuple (same_name_dict, same_description_dict)        The intrepretation of the results are handled in the main Insert body. Refer
    to inline comments for flow.

codyleyhan committed:
Added auth to restaurants

Commit:
Merge branch 'item' into itemMaster

Commit:
deleted import pprint

Commit:
Merge branch 'restaurant' of https://github.com/codyleyhan/dinero into restaurant

Pull information from cody's branch into my branch so I can see what is goin
on with it all

Commit:
added TODO's

codyleyhan committed:
Fixed code

Commit:
aliased dbs = db.session and rollback() implemented into excepts.

Commit:
in Update() code from if form.validate()'s else into construct_return_package()

Commit:
Up to if condition altered

Commit:
changed add() validation scheme to mimic update item scheme. added __eq__ to class Item

Commit:
got *iid working

Commit:
Good before altering items_with_same algorithm

Commit:
identifying duplicate items works. ex takes 3 item list to 2 items by removing duplicate items. Rest of module not updated

Commit:
items_with_same() fully functions. rest of controller.py not updated. __eq__ now omits id checking

Commit:
duplicate item works.

Commit:
full duplicate item support enabled

Commit:
all function appear to work

Commit:
return message updated

Commit:
removed __eq__

Commit:
Merge branch 'restaurant' of https://github.com/codyleyhan/dinero into restaurant
Updating branch `restaurant` with updated versions, if any, supplied by cody

codyleyhan committed:
queries user from token on every request

Commit:
added 5 fields to item class, category is only live one at the moment.

Commit:
added class Category(Enum) in constants.py

Commit:
Category added

Commit:
delete all functionallity added

codyleyhan committed:
Merge branch 'master' into restaurant

Commit:
testFuncs.py added, and successfully exectues

Commit:
Makeshift changes made here. It works, but there are a few restrictions.
First, for index, indux, at this point, must be accompanied by a string.
command = "test" means we get a print out of the class representation.
command = "cust" means we get the customers view of the class (using str)

similar makeshift changes made to update also. method= 'DELETE', itemId= 0
=> delete all items in the database.

Commit:
Merge branch 'restaurant' of https://github.com/codyleyhan/dinero into restaurant
Pull all changes from cody

Commit:
update from restaurant branch

Commit:
removed diff markers

Commit:
added user auth. has NOT been debugged

Commit:
auth add working correctly for admin status

Commit:
Added funcationality to .route(..parameters..)
renamed several items thoughout the body
added comments to improve code readability

Commit:
removed def delete_all()

Commit:
error removed

Commit:
stash save

Commit:
added comma after 'manager' in /aut/constants.py

Commit:
tweeks. not run yet

Commit:
edits implemented for pull request

Commit:
With Autorizatoin 'ADMIN', Known working funcationality:
	POST	/item/test
	GET	/item
	DELETE	/item/delete-all
	DELETE	/item/<int:itemId>

Commit:
Merge branch 'updateItemCategories' into item
pulling informatoin from masterg
